src/main/java/com/conveyal/r5/profile/PerTargetPropagater.java

```diff
@@ -214,23 +222,25 @@ private void propagateTransit (int targetIndex) {
         // the reducer later in the caller, because you can walk even where there is no transit.
         if (pointToStopDistanceTable != null) {
             pointToStopDistanceTable.forEachEntry((stop, distanceMillimeters) -> {
-                for (int iteration = 0; iteration < nIterations; iteration++) {
-                    int timeAtStop = travelTimesToStop[stop][iteration];
-                    if (timeAtStop > cutoffSeconds || timeAtStop > perIterationTravelTimes[iteration]) {
-                        // Skip propagation if all resulting times will be greater than the cutoff and
-                        // cannot improve on the best known time at this iteration. Also avoids overflow.
-                        continue;
-                    }
-                    // If recording path details, extract the row of paths to all stops for this iteration.
-                    // Propagate from the current stop out to the target.
-                    int timeAtTarget = timeAtStop + distanceMillimeters / speedMillimetersPerSecond;
-                    if (timeAtTarget < cutoffSeconds &&
-                        timeAtTarget < perIterationTravelTimes[iteration]) {
-                        // To reach this target, alighting at this stop is faster than any previously checked stop.
-                        perIterationTravelTimes[iteration] = timeAtTarget;
-                        if (calculateComponents) {
-                            Path[] pathsToStops = pathsToStopsForIteration.get(iteration);
-                            perIterationPaths[iteration] = pathsToStops[stop];
+                if (distanceMillimeters / speedMillimetersPerSecond <= egressLegTimeLimitSeconds) {
```

Comment:
Thanks for pointing out the core change. So if I understand right we were not applying distance limits to walking access legs, but are now; we were already applying distance limits to transfers but it's been refactored for clarity; and were not applying distance limits to egress legs but are now. These distance limits can only shorten the egress distances, making them less than the maximum length used when building the distance tables. That might surprise people in some cases but I guess it's better than not applying any limits at all.

We should probably refactor to pull this division outside the foreach loop - division is a slow operation and I don't know if the JVM can figure out that these variables won't change on each loop iteration, especially since the loop body is a lambda function call. That would mean applying an egress distance limit rather than an egress time limit.